packages/server/backup/scripts/restore_app.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+# Usage: ./restore_app.sh <backup_file.tar.gz>
+
+if [ -z "$1" ]; then
+  echo "ERROR: Please specify backup file (e.g., app_backup_20240501_020000.tar.gz)"
+  exit 1
+fi
+
+BACKUP_FILE="/backups/$1"
+APP_DIR="/app"  # Mount point in container
+
+mkdir -p "$APP_DIR"
+
+if [ ! -f "$BACKUP_FILE" ]; then
+  echo "ERROR: Backup file $BACKUP_FILE not found!"
+  exit 1
+fi
+
+echo "=== Restoring application from $1 ==="
+echo "This will OVERWRITE existing files. Continue? (y/N)"
+read -r confirm
+
+if [ "$confirm" != "y" ]; then
+  echo "Cancelled."
+  exit 0
+fi
+
+# Extract with permissions preserved
+tar -xzvf "$BACKUP_FILE" -C "$APP_DIR" --strip-components=1
```

Comment:
Before decompressing the backup into the app folder, make a backup of the current app folder before the restoration, with a name which mention it's being created during a restoration process, if it's not too much hassle